test/functional/connection/connection.ts

```diff
@@ -359,5 +359,33 @@ describe("Connection", () => {
         })));
 
     });
+
+    describe("Can change postgres default schema name", () => {
+        let connections: Connection[];
+        beforeEach(async () => {
+            connections = await createTestingConnections({ 
+                enabledDrivers: ["postgres"],
+                entities: [Post]
+            });
+        });
+        afterEach(() => closeTestingConnections(connections));        
+        it("schema name can be set", () => {
+            return Promise.all(connections.map(async connection => {
+                await connection.syncSchema(true);
+
+                const post = new Post();
+                post.title = "ChangeSchemaName";
+
+                const PostRepo = connection.getRepository(Post);
+                await PostRepo.persist(post);
+
+                const query = await connection.driver.createQueryRunner();
+                const schemaName = connection.driver.options.schemaName || "public";
```

Comment:
but here in your test it will be always "public", because connection.driver.options.schemaName is not set because you are enabling wrong driver. just console.log(schemaName); and you'll see it